### DISCUSSION TITLE
Vscode support

### DISCUSSION THREAD

**Discussion by flapjack-sdk** [shellhub-io/shellhub#2322] [Q&A] [answered]
Dear Shellhub Team

What a wonderful project!  Just out of curiosity is there some Vscode Remote Session support planned?.
At the moment it seems that opening a remote session via Vscode using shellhub does not seem to work due to
tunneling issues. Would you possible be so kind as to comment if this?

Vscode: Version: 1.74.2, OS: Linux x64 4.19.0-22-amd64 snap
Vscode->connectToHost (RemoteSSH)
[Error] Failed to set up socket for dynamic port forward to remote port 42833: Socket closed. Is the remote port correct?

Best Regards

Matthias

**Comment by gustavosbarreto** [comment]
Hi @flapjack-sdk

> What a wonderful project!

Thanks 😊 

> Just out of curiosity is there some Vscode Remote Session support planned?.

Currently, there is no plan to support it but it is a good idea. We just don't have the time and knowledge required of how vscode works to do that.

> At the moment it seems that opening a remote session via Vscode using shellhub does not seem to work due to
tunneling issues.

Yep. Tunneling doesn't work very well yet.

**Comment by gustavosbarreto** [comment, marked as answer]
@flapjack-sdk,

We're happy to inform you that the support for Vscode Remote Session was implemented in our latest stable version. You should now be able to open a remote session via Vscode using Shellhub without any issues.

We appreciate your suggestion and hope that you continue to enjoy using Shellhub. If you have any further questions or feedback, please don't hesitate to reach out to us.

**Reply by flapjack-sdk** [reply]
Dear Gstavosbarreto

What wonderful progress! We tested it and it's working. 
Now shellhub is becoming really productive for remote working!

Cheers!

Matthias
